Sources/FoundationEssentials/Calendar/Calendar_Recurrence.swift

```diff
@@ -232,6 +235,8 @@ extension Calendar {
                 }
                 var componentsForEnumerating = recurrence.calendar._dateComponents(components, from: start) 
                 
+                startDateNanoseconds = start.timeIntervalSince1970.remainder(dividingBy: 1)
```

Comment:
Would it work if we modify `components` above to include nanoseconds so the information is readily available in `componentsForEnumerating`, so that we don't have to do this manual adjustment here?

Reply:
@itingliu that was my first attempt, but it doesn't work.

in my understanding, the issue is that `Calendar.DatesByRecurring`'s underlying `baseRecurrence` (`Calendar.DatesByMatching`) will always return only whole-second-rounded dates (though this isn't documented, afaict), so we'd need to adjust that type's behaviour instead (which i decided against bc i wasn't sure whether this would affect any other dependent code, and i figured instead going with the simplest possible fix would be the best approach)